HeartbeatLogging/Sources/Storage.swift

```diff
@@ -70,11 +56,23 @@ extension FileStorage: PersistentStorage {
       throw StorageError.writeError
     }
   }
+
+  private func createDirectories(in url: URL) throws {
+    do {
+      try fileManager.createDirectory(
+        at: url,
+        withIntermediateDirectories: true,
+        attributes: nil
+      )
+    } catch CocoaError.fileWriteFileExists {
```

Comment:
I like Swift error handling so much more than Objective-C one!

@ncooke3 @paulb777 I think improving Firebase error system to optimize for Swift handling can be a next significant API improvement.